src/engine/engineworkerscheduler.cpp

```diff
@@ -45,15 +42,17 @@ void EngineWorkerScheduler::runWorkers() {
 void EngineWorkerScheduler::run() {
     while (!m_bQuit) {
         Event::start("EngineWorkerScheduler");
-        EngineWorker* pWorker = NULL;
-        while (m_scheduleFIFO.read(&pWorker, 1) == 1) {
-            if (pWorker) {
+        m_mutex.lock();
+        for(const auto& pWorker: m_workers) {
+            if (pWorker->isReady()) {
```

Comment:
The worker should decide internally if it should actually wake up when triggered. With this modification std::atomic_flag looks like a perfect choice for m_ready.